.github/actions/create-test/action.yml

```diff
@@ -0,0 +1,100 @@
+# Copyright 2024 The Ground Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Create a new survey
+
+inputs:
+  platform-repository:
+    description: 'ground-platform repository under test'
+    default: google/ground-platform
+
+  upload-artifacts:
+    description: 'Whether to upload the final emulator data artifacts'
+    default: 'false'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.platform-repository }}
+
+    - name: Load cached Firebase emulator
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/firebase/emulators
+        key: ${{ runner.os }}-firebase-emulators-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-firebase-emulators-
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: "18"
+
+    - name: Load cached Node modules
+      id: cache-npm
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-node-modules
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+      name: List the state of node modules
+      continue-on-error: true
+      shell: bash
+      run: npm list
+
+    - name: Install Chrome
+      uses: browser-actions/setup-chrome@v1
+      id: setup-chrome
+      with:
+        chrome-version: 125
+    
+    - name: Save CHROME_PATH
+      shell: bash
+      run: echo "CHROME_PATH=${{ steps.setup-chrome.outputs.chrome-path }}" >> $GITHUB_ENV
+
+    - name: Install nx-linux-x64-gnu
+      shell: bash
+      run: npm install @nx/nx-linux-x64-gnu
+
+    - name: Install dependencies
+      shell: bash
+      run: npm run build:local
+
+    - name: Run create e2e tests
```

Comment:
Nit: At first glance I think this would actually create the tests - perhaps we can just call these various steps "step 1, 2," etc, or refer to them by a more complete name like "'Create survey' test"?

Reply:
Done for both files.